Feature/teen: TeenView / TeenDetailView UI 구현

ha-nabi opened the pull request:
## PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 🔎 작업 내용

- Teen 메인 뷰 UI 작업

  <br/>

## 이미지 첨부

https://github.com/user-attachments/assets/09e7b280-19c7-497d-a5fa-7c01c758263e

<br/>

## ➕ 이슈 링크
- #40

<br/>

hamfan524 committed:
🩹 Refactor: ReportFeature 분리

MainFeature 내에 있던 신고 모달 기능을 ReportFeature로 분리해서 TeenFeature에서도 접근 가능하게 구현

Co-Authored-By: Noh Joo Young <96951827+JooYoungNoh@users.noreply.github.com>

hamfan524 committed:
🏫 Feat: Teen - 상세 페이지 구현

MainFeature에서 사용하던 TeenTableViewCell을 DesignSystem으로 추가하여 여러 뷰에서 재사용하도록 수정

hamfan524 committed:
Merge branch 'develop' into feature/Teen

hamfan524 committed:
🎸 Chore: Develop브랜치에서 merge

Teen관련 셀 업데이트 사항 충돌난부분 처리

ha-nabi committed:
🏫 Feat: Teen - Teen 메인 화면 구현

버튼 액션 연결 X

ha-nabi committed:
🏫 Feat: Teen - Teen 버튼 액션 연결

ha-nabi committed:
🏫 Feat: Teen - Teen 메인 화면 셀 수정

hamfan524 committed:
🩹 Refactor: Teen/Login - Memory Leak 현상 해결

TeenDetail -> 강하게 참조중이던 클로저 값들을 약한참조로 변경하여 해결

Login -> 시트가 닫힐때, RootNavigationController의 하위에 있는 코디네이터와 컨트롤러가 해제되지 않는 문제가 있어 해결
- 내부 커스텀 서브뷰들도 모두 해제하도록 수정

Co-Authored-By: Noh Joo Young <96951827+JooYoungNoh@users.noreply.github.com>
Co-Authored-By: Lee ChangJun <109324421+Phangg@users.noreply.github.com>

ha-nabi committed:
🏫 Feat: Teen - Teen 페이징 인디케이터 구현

디자인 수정 해야함

hamfan524 committed:
🩹 Refactor: 페이징 인디케이터 재사용하도록 수정

ProfileDetail / Teen 에서 사용하는CustomPageControlView를 생성자에 가로와 세로길이를 받으면서 다른 프로젝트에서도 재사용하도록 수정

+ DesignSystem 프로젝트로 이동

hamfan524 committed:
🐞 Fix: 신고 후 알럿 다이얼로그 안나오던 오류 수정

JooYoungNoh committed:
🐞 Fix: 메인 탭 다시 빌드하는 오류 임시 수정

- 메인탭 재실행 -> Notification으로 바꿔봄

ha-nabi committed:
🩹 Refactor: 셀을 눌렀을 때 셀의 label이 디테일 뷰의 label이랑 연동되게 구현

ha-nabi committed:
🩹 Refactor: with 파라미터 제거

ha-nabi committed:
Teen / Ranking의 String 값들 ViewValue, DesignSystem 이동

ha-nabi committed:
Merge branch 'develop' into feature/Teen

ha-nabi committed:
🐞 Fix: 중복 코드 제거

ha-nabi committed:
🩹 Refactor: TeenViewModel의 labels 수정

디테일 뷰에선 \n (줄바꿈)을 제거

ha-nabi committed:
🏫 Feat: TeenDetailView에서 TeenView로 뒤돌아 갈 때 스와이핑 활성화

hamfan524 reviewed (approved):
고생하셨습니다! 
혹시 피그마랑 Teen 화면의 레이아웃 값이 다른건 기기에 따라 다르게 처리된건가요??

Phangg reviewed (commented):
@ha-nabi 
궁금한게 있어서 질문 2개 드려요!!

1. "드래그 제스처를 통한 뒤로가기" 는 혹시 회의를 통해서 정해진건가요?
그게 맞으면 다른 화면들에도 적용을 하는게 좋을 것 같아서 확인차 물어봅니다!

2. 제가 올린 이미지의 빨간색 부분의 padding 이 디자인과 차이가 꽤 나는 것 같아요
혹시 뭐 다른 부분이 있을까요?
<img width="384" alt="스크린샷 2024-07-29 오후 3 21 10" src="https://github.com/user-attachments/assets/5e257ba8-25b1-4162-bf0c-1a34edf777b1">

devhanavi commented:
> 고생하셨습니다! 혹시 피그마랑 Teen 화면의 레이아웃 값이 다른건 기기에 따라 다르게 처리된건가요??

Teen 레이아웃이 다른건 피그마는 스크롤뷰 형식으로 되어있어서 셀 크기가 달라서 그렇습니다
지금은 현재 좌, 우로 디폴트 패딩이 적용되어 있습니다

ps. 저번에 물어봤을 때 페이징 했을 때 추가적인 셀이 보인다는걸 알려주고 있는거고 작업은 페이징으로 하자고 했던거 같은데 수정사항 있으면 수정하겠습니다

devhanavi commented:
> 1. "드래그 제스처를 통한 뒤로가기" 는 혹시 회의를 통해서 정해진건가요?
>    그게 맞으면 다른 화면들에도 적용을 하는게 좋을 것 같아서 확인차 물어봅니다!
> 2. 제가 올린 이미지의 빨간색 부분의 padding 이 디자인과 차이가 꽤 나는 것 같아요
>    혹시 뭐 다른 부분이 있을까요?
> 
> <img alt="스크린샷 2024-07-29 오후 3 21 10" width="384" src="https://private-user-images.githubusercontent.com/109324421/352936031-5e257ba8-25b1-4162-bf0c-1a34edf777b1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyMzQ2ODEsIm5iZiI6MTcyMjIzNDM4MSwicGF0aCI6Ii8xMDkzMjQ0MjEvMzUyOTM2MDMxLTVlMjU3YmE4LTI1YjEtNDE2Mi1iZjBjLTFhMzRlZGY3NzdiMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyOVQwNjI2MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YmE1N2UxOWViOWI5ZDk2OTMwNWRiMWViZDFkMjY0ZGI4OTgzZTc2YjNlZDg0MGU0M2IxNDdmYWJmNjRhNGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.VilFRwVu5Xk6zTYM3AQE7Rq33PYRbYgDH3WJhfRDwW4">

엇 드래그제스처를 막자는 소리가 있었나요?
당연한건줄 알고 활성화 시킨건데 기기 테스트 할 때 제스처 안되면 생각보다 불편한 점이 많아서 저는 꼭 활성화 해두는 편이긴 합니다만 나중에 한번 얘기해보죠 ㅎ.ㅎ

디테일 뷰는 동호님이 ui작업한거라 신경 못쓰고있었는데 지금 보니 그렇네요 !
제가 수정해놓겠습니다

Phangg reviewed (approved):
@ha-nabi 

네네! 확인 감사합니다. 
저도 뒤로가기는 각 화면 성격에 맞춰서 필요한 부분에 적용하는 것도 좋다고 생각해요! ㅎㅎ

고생하셨습니다 😆

ha-nabi committed:
🩹 Refactor: 디테일 뷰 패딩 안맞는 문제 해결

hamfan524 committed:
🐞 Fix: 뒤로가기를 한뒤 시트를 닫으면 메모리에서 할당해제 안되는 오류 수정

PopViewController 호출 시 dismissNavigation 클로저 nil로 변경되는 부분 발견

-> Sheet가 dismiss될 때만 dismissNavigation를 nil로 변경하도록 수정

Co-Authored-By: Lee ChangJun <109324421+Phangg@users.noreply.github.com>
Co-Authored-By: ciu <a2849535@gmail.com>

hamfan524 committed:
💄 Design: 앱 아이콘 적용

Co-Authored-By: Lee ChangJun <109324421+Phangg@users.noreply.github.com>
Co-Authored-By: ciu <a2849535@gmail.com>
